### PR TITLE
Feature/asset list tokens tab

### PR DESCRIPTION
**Pull request by svojsu**
- Introduce new grouping logic for asset groups in `AssetListBaseBuilder`.
- Refactor `AssetListAssetViewModelFactory` to support the new groups.
- Introduce `AssetListCollectionManager` to separate collection view management from controller logic.
- Introduce `AssetListNetworksFlowLayout` and `AssetListTokensFlowLayout`.
- Temporarily disable search logic.
- Temporarily use the filter button to swap layouts.

### CODE REVIEW

**Review by ERussel — COMMENTED** [review]
1) I think we need to get rid of changes for assets and chains for now as current implementation significantly complicates the logic. So, I would just rebuild all the groups on chain update/insert/delete. On balance update we can still update only those assets for which balance got updated.

2) 1 also fixes the issue with assets enabled/disabled since we rebuild the groups

3) Also noticed that we don't apply hide zero balances to the group by asset in the presenter

**Comment by svojsu** [comment]
> 1. I think we need to get rid of changes for assets and chains for now as current implementation significantly complicates the logic. So, I would just rebuild all the groups on chain update/insert/delete. On balance update we can still update only those assets for which balance got updated.
> 2. 1 also fixes the issue with assets enabled/disabled since we rebuild the groups
> 3. Also noticed that we don't apply hide zero balances to the group by asset in the presenter

Builder:

1. Removed the mapping of changes, now we rebuild the groups with lists. Left the changes in balances.
2. Using `MultichainAsset` instead of `ChainAsset` for asset groups.
3. Removed calculators from the model, added them yesterday, and removed the group diffs from the model today.
4. Cleared all unnecessary things from the helpers that were needed for mapping changes.

Presenter:

1. Handled the filter with `hidesZeroBalances`.
2. In the view model, I use balances from the group, not the sum of asset balances.

File: novawallet/Modules/AssetList/Base/AssetListAssetsViewModelFactory.swift [hunk 1]
```diff
@@ -35,17 +52,45 @@ class AssetListAssetViewModelFactory {
...
         currencyManager: CurrencyManagerProtocol
     ) {
+        self.chainAssetViewModelFactory = chainAssetViewModelFactory
         self.priceAssetInfoFactory = priceAssetInfoFactory
         self.assetFormatterFactory = assetFormatterFactory
         self.percentFormatter = percentFormatter
         self.currencyManager = currencyManager
     }
 
+    func createBalanceViewModel(
```
[the remark this answers is not in the document]
Comment (svojsu) [reply novawallet/Modules/AssetList/Base/AssetListAssetsViewModelFactory.swift:68]: Actually, we need this method, but reimplemented for groups so we don't calculate the sum again.

File: novawallet/Modules/AssetList/Models/AssetListBuilderResult.swift [hunk 2]
```diff
@@ -1,10 +1,30 @@
 import Foundation
 import BigInt
+import Operation_iOS
+
+extension ListDifferenceCalculator where T == AssetListChainGroupModel {
```
[the remark this answers is not in the document]
Comment (svojsu) [reply novawallet/Modules/AssetList/Models/AssetListBuilderResult.swift:5]: We don't need it anymore.

File: novawallet/Modules/AssetList/View/AssetListCollectionManager/AssetListCollectionViewDataSource.swift [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+            for: .touchUpInside
+        )
+
+        return settingsCell
+    }
+
+    func provideAssetCell(
+        _ collectionView: UICollectionView,
+        indexPath: IndexPath,
+        assetIndex _: Int
```
Comment (ERussel) [inline novawallet/Modules/AssetList/View/AssetListCollectionManager/AssetListCollectionViewDataSource.swift:134]: Not used assetIndex

**Review by ERussel — APPROVED** [review]

**Pull request merged**